--- /etc/rc.d/rc.S.dist	2013-10-28 06:06:39.000000000 +0100
+++ /etc/rc.d/rc.S	2014-04-02 10:06:36.000000000 +0200
@@ -4,6 +4,8 @@
 #
 # Mostly written by:  Patrick J. Volkerding, <volkerdi@slackware.com>
 #
+# This version has been updated to run inside a Lxc container!
+container="lxc"
 
 PATH=/sbin:/usr/sbin:/bin:/usr/bin
 
@@ -19,6 +21,9 @@
   fi
 fi
 
+# Disabled inside the container
+if [ ! $container = "lxc" ]; then
+
 # If /run exists, mount a tmpfs on it (unless the
 # initrd has already done so):
 if [ -d /run ]; then
@@ -270,13 +275,17 @@
   echo -n "Press ENTER to continue. "
   read junk;
 fi # Done checking root filesystem
+fi
+# end container
 
 
 # Any /etc/mtab that exists here is old, so we start with a new one:
 /bin/rm -f /etc/mtab{,~,.tmp} && /bin/touch /etc/mtab
 
+# Disabled inside the container
 # Add entry for / to /etc/mtab:
-/sbin/mount -f -w /
+#/sbin/mount -f -w /
+# end container
 
 # Add /proc and /sys mounts to /etc/mtab:
 if [ -d /proc/sys ]; then
@@ -286,6 +295,9 @@
   /sbin/mount -f sysfs /sys -t sysfs
 fi
 
+# Disabled inside the container
+if [ ! $container = "lxc" ]; then
+
 # Configure ISA Plug-and-Play devices:
 if [ -r /etc/isapnp.conf ]; then
   if [ -x /sbin/isapnp ]; then
@@ -318,6 +330,8 @@
 if [ -x /sbin/sysctl -a -r /etc/sysctl.conf ]; then
   /sbin/sysctl -e -p /etc/sysctl.conf
 fi
+fi
+# end container
 
 # Check all the non-root filesystems:
 if [ ! -r /etc/fastboot ]; then
@@ -349,10 +363,12 @@
   /sbin/mount -a -v -t nonfs,nosmbfs,nocifs,noproc,nosysfs
 fi
 
+# # Disabled inside the container
 # Enable swapping again.  This is needed in case a swapfile is used,
 # as it can't be enabled until the filesystem it resides on has been
 # mounted read-write.
-/sbin/swapon -a 2> /dev/null
+#/sbin/swapon -a 2> /dev/null
+# end container
 
 # Clean up some temporary files:
 rm -f /var/run/* /var/run/*/* /var/run/*/*/* /etc/nologin \
@@ -377,25 +393,30 @@
 chown root:utmp /var/run/utmp
 chmod 664 /var/run/utmp
 
+# Modified inside the container
 # Update the current kernel level in the /etc/motd (Message Of The Day) file,
 # if the first line of that file begins with the word 'Linux'.
 # You are free to modify the rest of the file as you see fit.
 if [ -x /bin/sed ]; then
-  /bin/sed -i "{1s/^Linux.*/$(/bin/uname -sr)\./}" /etc/motd
+  #/bin/sed -i "{1s/^Linux.*/$(/bin/uname -sr)\./}" /etc/motd
+  /bin/sed -i "{1s/^Linux.*/$(/bin/uname -sr) lxc container\./}" /etc/motd
 fi
+# end container
 
 # If there are SystemV init scripts for this runlevel, run them.
 if [ -x /etc/rc.d/rc.sysvinit ]; then
   . /etc/rc.d/rc.sysvinit
 fi
 
+# # Disabled inside the container
 # Run serial port setup script:
 # CAREFUL!  This can make some systems hang if the rc.serial script isn't
 # set up correctly.  If this happens, you may have to edit the file from a
 # boot disk, and/or set it as non-executable:
-if [ -x /etc/rc.d/rc.serial ]; then
-  sh /etc/rc.d/rc.serial start
-fi
+#if [ -x /etc/rc.d/rc.serial ]; then
+#  sh /etc/rc.d/rc.serial start
+#fi
+#end container
 
 # Carry an entropy pool between reboots to improve randomness.
 if [ -f /etc/random-seed ]; then
